rules/adcrule.go

```diff
@@ -0,0 +1,151 @@
+//Package rules ...
+// Copyright (c) 2017 Intel Corporation.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package rules
+
+import (
+	"fmt"
+	"github.com/intel-go/nff-go/common"
+	"github.com/intel-go/nff-go/flow"
+	"gopkg.in/ini.v1"
+	"net"
+	"os"
+	"sort"
+	"strconv"
+)
+
+//ADCFilter ...
+type ADCFilter struct {
+	ADCFilterIdx int
+	ADCType      int //: 1-IP, 2-IP+Prefix, 3-Domain name
+
+	IP     string //(IP Address)
+	Prefix int
+
+	IPPrefixNW net.IPNet //lookup ease
+
+	Domain string
+}
+
+//ParseADCRules ...
+func ParseADCRules(filepath string) []ADCFilter {
+	common.LogInfo(common.Info, "Loading ADC Rules...")
+	cfg, err := ini.Load(filepath)
+	flow.CheckFatal(err)
+
+	noOfFilters, err := cfg.Section(LblGlobal).Key(LblNumAdcRules).Int()
+	flow.CheckFatal(err)
+
+	for i := 1; i <= noOfFilters; i++ {
+		strVal := strconv.Itoa(i)
+		section := cfg.Section(LblAdcRule + "_" + strVal)
+
+		adcFilter := ADCFilter{ADCFilterIdx: i}
+
+		if section.HasKey(LblAdcType) {
+			adcFilter.ADCType, err = section.Key(LblAdcType).Int()
+			flow.CheckFatal(err)
+		}
+
+		if section.HasKey(LblIP) {
+			adcFilter.IP = section.Key(LblIP).String()
+
+		}
+		if section.HasKey(LblPrefix) {
+			adcFilter.Prefix, err = section.Key(LblPrefix).Int()
+			adcFilter.IP = section.Key(LblIP).String()
+
+			_, ipnet, err := net.ParseCIDR(adcFilter.IP + "/" + strconv.Itoa(adcFilter.Prefix))
+			flow.CheckFatal(err)
+
+			adcFilter.IPPrefixNW = *ipnet //ip-net will ease the lookup
+		}
+
+		if section.HasKey(LblDomain) {
+			adcFilter.Domain = section.Key(LblDomain).String()
+
```

Comment:
Please, use "gofmt -s -w" for all .go files

Reply:
i used gofmt -w option may be i missed it let me do it again. I think it's better to integrate gofmt and golint in makefile. so, build itself will catch it.

Reply:
Anyway, remove empty lines before closing brackets